File: CIS/matlab_scripts/cal_steng.m
function [out]= cal_steng (sig, Nlen, Nshift)


% k*Nshift+1:k*Nshift+Nlen

% ind=1:Nshift:length(sig)-2*Nlen;
% indall=[ind];
% for j=1:Nlen-1
%     indall=[indall; indall(end,:)+1];
% end
% sig_seg=sig(indall);
% out=mean(sig_seg.^2,1);

out1=filtfilt(ones(1,Nlen)/100,1,sig.^2);
out=out1(1:Nshift:end);

File: python_scripts/utils/cal_steng.py

#function [out]= cal_steng (sig, Nlen, Nshift)


#out1=filtfilt(ones(1,Nlen)/100,1,sig.^2);
#out=out1(1:Nshift:end);

from scipy.signal import filtfilt
import numpy as np

def cal_steng(sig, Nlen, Nshift):
    out1= filtfilt(np.ones((np.int(Nlen),))/100,1,sig**2)
    #Considers elements from 1 to end with a shift of Nshift
    out=out1[::Nshift]
    return out
